test/odbc/psqlodbc/bytea.cpp

```diff
@@ -0,0 +1,377 @@
+    pg_test_setup.ConnectAndExecQuery(DropObjectStatement("TABLE", PG_TABLE_NAME));
+  }
+
+  void TearDown() override {
+    if (!Drivers::DriverExists(ServerType::PSQL)) {
+      GTEST_SKIP() << "PSQL Driver not present: skipping all tests for this fixture.";
+    }
+    if (!Drivers::DriverExists(ServerType::MSSQL)) {
+      GTEST_SKIP() << "MSSQL Driver not present: skipping all tests for this fixture.";
+    }
+
+    OdbcHandler bbf_test_setup(Drivers::GetDriver(ServerType::MSSQL));
+    bbf_test_setup.ConnectAndExecQuery(DropObjectStatement("VIEW", BBF_VIEW_NAME));
+    bbf_test_setup.CloseStmt();
+    bbf_test_setup.ExecQuery(DropObjectStatement("TABLE", BBF_TABLE_NAME));
+    
+    OdbcHandler pg_test_setup(Drivers::GetDriver(ServerType::PSQL));
+    pg_test_setup.ConnectAndExecQuery(DropObjectStatement("VIEW", PG_VIEW_NAME));
+    pg_test_setup.CloseStmt();
+    pg_test_setup.ExecQuery(DropObjectStatement("TABLE", PG_TABLE_NAME));
+  }
+};
+
+string stringToHex(string input) {
+    std::ostringstream result;
+    result << std::setw(2) << std::setfill('0') << std::hex << std::uppercase;
+    std::copy(input.begin(), input.end(), std::ostream_iterator<unsigned int>(result, ""));
+    return result.str();   
+}
+
+vector<string> getExpectedResults(const vector<string> &input) {
+  vector<string> ret = {};
+
+  for (int i = 0; i < input.size(); i++) {
+    ret.push_back(stringToHex(input[i]));
+  }
+
+  return ret;
+}
+
+vector<string> duplicateElements(vector<string> input) {
+  typedef std::move_iterator<decltype(input)::iterator> VecMoveIter;
+  std::vector<string> duplicated(input);
+  std::copy(VecMoveIter(input.begin()), VecMoveIter(input.end()), std::back_inserter(duplicated));
+  return duplicated;
+}
+
+TEST_F(PSQL_DataTypes_ByteA, Table_Creation) {
+  createTable(ServerType::MSSQL, BBF_TABLE_NAME, TABLE_COLUMNS);
+
+  const vector<int> BBF_LENGTH_EXPECTED = {10, 0};
+  const vector<int> BBF_PRECISION_EXPECTED = {10, 0};
+  const vector<int> BBF_SCALE_EXPECTED = {0, 0};
+  const vector<string> BBF_NAME_EXPECTED = {"int", "varbinary"};
+
+  testCommonColumnAttributes(ServerType::MSSQL, BBF_TABLE_NAME, 
+    TABLE_COLUMNS.size(), COL1_NAME, 
+    BBF_LENGTH_EXPECTED, BBF_PRECISION_EXPECTED, 
+    BBF_SCALE_EXPECTED, BBF_NAME_EXPECTED);
+
+  const vector<int> PG_LENGTH_EXPECTED = {4, -4};
+  const vector<int> PG_PRECISION_EXPECTED = {0, 0};
+  const vector<int> PG_SCALE_EXPECTED = {0, 0};
+  const vector<string> PG_NAME_EXPECTED = {"int4", "bytea"};
+
+  testCommonColumnAttributes(ServerType::PSQL, PG_TABLE_NAME, 
+    TABLE_COLUMNS.size(), COL1_NAME, 
+    PG_LENGTH_EXPECTED, PG_PRECISION_EXPECTED, 
+    PG_SCALE_EXPECTED, PG_NAME_EXPECTED);
+
+  dropObject(ServerType::PSQL, "TABLE", PG_TABLE_NAME);
+  dropObject(ServerType::MSSQL, "TABLE", BBF_TABLE_NAME);
+}
+
+TEST_F(PSQL_DataTypes_ByteA, Insertion_Success) {
+  vector<string> inserted_values = {
+    "NULL",
+    "",
+    "A",
+    "123"
```

Comment:
From the definition:
> A binary string is a sequence of octets (or bytes). Binary strings are distinguished from character strings in two ways. First, binary strings specifically allow storing octets of value zero and other “non-printable” octets (usually, octets outside the decimal range 32 to 126).

I don't see the case of handling values outside of 32 to 126. Could we add that in?

Reply:
Added.